crates/evm-prover/src/prover/block.rs

```diff
@@ -425,7 +425,12 @@ impl BlockExecProver {
             .share_get_namespace_data(&header_clone, self.app.namespace)
             .await?;
 
-        let proofs: Vec<NamespaceProof> = namespace_data.rows.iter().map(|row| row.proof.clone()).collect();
+        let proofs: Vec<NamespaceProof> = namespace_data
+            .rows
+            .iter()
+            .filter(|row| row.proof.is_of_presence())
+            .map(|row| row.proof.clone())
```

Comment:
minor idiomatic: https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.filter_map

Reply:
It looks like clippy prefers `filter` + `map` in this case:
```
error: usage of `bool::then` in `filter_map`
Error:    --> crates/evm-prover/src/prover/block.rs:431:14
    |
431 |             .filter_map(|row| row.proof.is_of_presence().then(|| row.proof.clone()))
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `filter` then `map` instead: `filter(|&row| row.proof.is_of_presence()).map(|row| row.proof.clone())`
```

Reply:
I reverted the commit